src/main/kotlin/com/hibob/academy/dao/PetDao.kt

```diff
@@ -1,2 +1,60 @@
 package com.hibob.academy.dao
 
+import jakarta.inject.Inject
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.RecordMapper
+
+class PetDao @Inject constructor(private val sql: DSLContext) {
+
+    private val pet = PetTable.instance
+
+    private val petMapper = RecordMapper<Record, Pet>
+    { record ->
+        Pet (
+            id = record[pet.id],
+            name = record[pet.name],
+            type = record[pet.type],
+            companyId = record[pet.companyId].toLong(),
+            arrivalDate = record[pet.dateOfArrival],
+            ownerId = record[pet.ownerId].toLong()
+        )
+    }
+
+    private val petWithoutTypeMapper = RecordMapper<Record, PetWithoutType>
+    { record ->
+        PetWithoutType (
+            record[pet.id],
+            record[pet.name],
+            record[pet.companyId].toLong(),
+            record[pet.dateOfArrival],
+            record[pet.ownerId].toLong()
+        )
+    }
+
+    fun getAllPetsFromType(type: PetType): List<PetWithoutType> =
+        sql.select(pet.name, pet.companyId, pet.dateOfArrival)
+            .from(pet)
+            .where(pet.type.eq(getPetType(type)))
+            .fetch(petWithoutTypeMapper)
```

Comment:
What is the difference between this function to the getAllPetsByType besides the mapper?